client/src/planwise/client/scenarios/handlers.cljs

```diff
@@ -133,10 +133,12 @@
  :scenarios/cancel-dialog
  in-scenarios
  (fn [db [_]]
-   (assoc db
-          :view-state :current-scenario
-          :changeset-dialog nil
-          :rename-dialog nil)))
+   (let [cancel-next-state (get-in db [:cancel-next-state])]
+     (assoc db
+            :view-state        (if (some? cancel-next-state) cancel-next-state :current-scenario)
```

Comment:
You can use `(or cancel-next-state :current-scenario)` here which should be equivalent in this case.